api/pkg/apis/projectcalico/v3/ippool.go

```diff
@@ -74,6 +74,10 @@ type IPPoolSpec struct {
 	// Allows IPPool to allocate for a specific node by label selector.
 	NodeSelector string `json:"nodeSelector,omitempty" validate:"omitempty,selector"`
 
+	// Allows IPPool to allocate for a specific namespace by label selector.
+	// If specified, both namespaceSelector and nodeSelector must match for the pool to be used.
+	NamespaceSelector string `json:"namespaceSelector,omitempty" validate:"omitempty,selector"`
```

Comment:
We should enhance the validation code to make sure this selector is valid.

https://github.com/projectcalico/calico/blob/291c8f5138429c8851bcbece1d5712fc1921567c/libcalico-go/lib/validator/v3/validator.go#L1127-L1247

It actually looks like we're missing validation logic for `NodeSelector` as well!